acurl/src/session.pyx

```diff
@@ -32,7 +32,7 @@ cdef BufferNode* alloc_buffer_node(size_t size, char *data):
     node.next = NULL
     return node
 
-cdef size_t header_callback(char *ptr, size_t size, size_t nmemb, void *userdata):
+cdef size_t header_callback(char *ptr, size_t size, size_t nmemb, void *userdata) noexcept:
```

Comment:
OTOH, this function (and the one below it in this file) really can't raise, and so are valid `noexcept` candidates with no explanation needed.